### PR TITLE
Fix isFinal for method infos.

### PR DESCRIPTION
**Pull request by brendandahl**

### CODE REVIEW

**Comment by brendandahl** [comment]
Removes 78 uneeded unwinds.

File: jit/analyze.ts [hunk 1]
```diff
@@ -90,7 +90,7 @@ module J2ME {
       return true;
     }
     // XXX Determine whether we can start using the code in this function.
-    return false;
+    return methodInfo.isFinal;
     var result = methodInfo.isFinal;
```
Comment (marco-c) [inline jit/analyze.ts]: Could we now use the code in this function? Maybe it wasn't working before a1fd332 and it's working now?

Comment (brendandahl) [inline jit/analyze.ts]: Not yet, that code is only valid if every class in every jar is loaded. However, mbx and I have been chatting on enabling better analysis.

Comment (marco-c) [inline jit/analyze.ts]: Ok, could you, just for clarity, comment it out?

**Comment by marco-c** [comment]
Looks good to me, just needs to be updated.

File: jit/analyze.ts [hunk 2]
```diff
@@ -89,26 +89,26 @@ module J2ME {
...
-              methodInfo.signature === subClassMethodInfo.signature) {
-            result = false;
-            break;
-          }
-        }
-      }
-    }
-    return result;
+    return methodInfo.isFinal;
+    // XXX The following can only be used if every class in all jars is loaded.
```
Comment (marco-c) [inline jit/analyze.ts]: Nit: An enclosing /* */ would have worked too :smile:

**Pull request merged**